@here/harp-mapview/lib/techniques/FillTechniqueHandler.ts

```diff
@@ -0,0 +1,137 @@
+import { assert } from "@here/harp-utils";
+import {
+    applyBaseColorToMaterial,
+    createMaterial,
+    getBufferAttribute
+} from "../DecodedTileHelpers";
+import { Tile } from "../Tile";
+import { techniqueHandlers, TechniqueUpdateContext, TileObjectEntry } from "./TechniqueHandler";
+import { GenericTechniqueHandler, TechniqueHandlerCommon } from "./TechniqueHandlerCommon";
+
+export class FillTechniqueHandler extends GenericTechniqueHandler<FillTechnique> {
+    mainMaterial: MapMeshBasicMaterial;
+
+    outlineMaterial?: EdgeMaterial;
+
+    mainObjects: TileObjectEntry[] = [];
+    outlineObjects: TileObjectEntry[] = [];
+
+    constructor(technique: FillTechnique, tile: Tile, context: TechniqueUpdateContext) {
+        // this.technique = fillDefaults(techniqe)
+        super(technique);
+        this.mainMaterial = createMaterial({ technique, env: context.env }) as MapMeshBasicMaterial;
+        assert(this.mainMaterial instanceof MapMeshBasicMaterial);
+
+        if (Expr.isExpr(technique.color) || Expr.isExpr(technique.opacity)) {
+            this.updaters.push(
+                TechniqueHandlerCommon.updateBaseColor.bind(
+                    undefined,
+                    this.mainMaterial,
+                    this.mainObjects,
+                    technique
+                )
+            );
+        }
+
+        const fadingParams = TechniqueHandlerCommon.getPolygonFadingParams(technique, context.env);
+        if (TechniqueHandlerCommon.isFadingFeatureEnabled(fadingParams)) {
+            this.updaters.push(
+                TechniqueHandlerCommon.updateFadingParams.bind(
+                    undefined,
+                    this.mainMaterial,
+                    fadingParams
+                )
+            );
+        }
+
+        if (
+            Expr.isExpr(technique.lineWidth) ||
+            (typeof technique.lineWidth === "number" && technique.lineWidth > 0)
+        ) {
+            const materialParams: EdgeMaterialParameters = {
+                color: fadingParams.color,
+                colorMix: fadingParams.colorMix,
+                fadeNear: fadingParams.lineFadeNear,
+                fadeFar: fadingParams.lineFadeFar
+            };
+            this.outlineMaterial = new EdgeMaterial(materialParams);
+
+            if (TechniqueHandlerCommon.isFadingFeatureEnabled(fadingParams)) {
+                this.updaters.push(
+                    TechniqueHandlerCommon.updateEdgeFadingParams.bind(
+                        undefined,
+                        this.outlineMaterial,
+                        fadingParams
+                    )
+                );
+            }
+
+            if (Expr.isExpr(technique.lineColor) || Expr.isExpr(technique.opacity)) {
+                this.updaters.push(this.updateLineColor.bind(this));
+            }
+        }
+    }
+
+    createObject(tile: Tile, srcGeometry: Geometry, group?: Group) {
```

Comment:
Not sure if this method should add the objects to the tile. Maybe it should just return the created objects and we explicitely add them in the TileGeometryCreator.
Or we should at least rename this method to addObjectsToTile

Reply:
Good point, i'll check if that simplify architecture.